implementations/rust/ockam/ockam_api/src/control_api/protocol/common.rs

```diff
@@ -1,8 +1,35 @@
 use ockam::identity::{Identity, Vault};
 use serde::{Deserialize, Serialize};
+use std::collections::BTreeMap;
 use std::str::FromStr;
 use utoipa::ToSchema;
 
+// This is an alias for documentation purposes only
+/// The destination node name.
+/// Depending on the type of node resolution used, it can be a relay name
+/// or a dns address.
+/// The special value `self` can be used to refer to the current node.
+#[derive(Serialize, Deserialize, ToSchema)]
+pub struct NodeName(String);
+
+// This is an alias for documentation purposes only
+#[derive(Serialize, Deserialize, ToSchema, Debug)]
+#[schema(
+    description =
+r#"
+Credential attributes.
+Attributes are key-value pairs that can be used to describe a credential.
+Ockam uses `ockam-` as a prefix for its own attributes.
+[You can learn more about attributes in the Ockam documentation](https://docs.ockam.io/reference/protocols/access-controls)
```

Comment:
I think that this page: https://docs.ockam.io/reference/protocols/access-controls, should make it clear what is pure data and what is library code:

 - Pure data is used on the command line and with the HTTP API.
 - Library code is used when developing your own node.

Reply:
I agree we need to polish/adjust some pages to have as an in-depth link, for this API, but also for examples and command help.